src/main/java/org/opensearch/neuralsearch/processor/CompoundTopDocs.java

```diff
@@ -113,7 +121,18 @@ private List<ScoreDoc> cloneLargestScoreDocs(final List<TopDocs> docs) {
                 maxScoreDocs = topDoc.scoreDocs;
             }
         }
+
         // do deep copy
-        return Arrays.stream(maxScoreDocs).map(doc -> new ScoreDoc(doc.doc, doc.score, doc.shardIndex)).collect(Collectors.toList());
+        List<ScoreDoc> scoreDocs = new ArrayList<>();
+        for (int scoreDocIndex = 0; scoreDocIndex < maxScoreDocs.length; scoreDocIndex++) {
+            if (isSortEnabled) {
+                FieldDoc fieldDoc = (FieldDoc) maxScoreDocs[scoreDocIndex];
+                scoreDocs.add(new FieldDoc(fieldDoc.doc, fieldDoc.score, fieldDoc.fields, fieldDoc.shardIndex));
+            } else {
+                ScoreDoc scoreDoc = maxScoreDocs[scoreDocIndex];
+                scoreDocs.add(new ScoreDoc(scoreDoc.doc, scoreDoc.score, scoreDoc.shardIndex));
+            }
+        }
```

Comment:
nit: move logic to build FieldDoc/ScoreDoc to separate method and call appropriately if sortEnabled or not. The above code will check for isSortEnabled every time, though, it is fixed for all the score docs.